internal/servers/controller/handlers/host_sets/host_set_service.go

```diff
@@ -539,6 +546,47 @@ func (s Service) updateInRepo(ctx context.Context, scopeId, catalogId string, re
 	return out, hl, nil
 }
 
+func (s Service) updatePluginInRepo(ctx context.Context, scopeId string, req *pbs.UpdateHostSetRequest) (host.Set, []host.Host, *plugins.PluginInfo, error) {
+	const op = "host_sets.(Service).updatePluginInRepo"
+	item := req.GetItem()
+	h, err := toStoragePluginSet(ctx, "", item)
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(ctx, err, op, errors.WithMsg("Unable to build host set for update"))
+	}
+	h.PublicId = req.GetId()
+	dbMask := pluginMaskManager.Translate(req.GetUpdateMask().GetPaths(), "attributes.")
+	if len(dbMask) == 0 {
+		return nil, nil, nil, handlers.InvalidArgumentErrorf("No valid fields included in the update mask.", map[string]string{"update_mask": "No valid fields provided in the update mask."})
+	}
+	repo, err := s.pluginRepoFn()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	out, hosts, plg, rowsUpdated, err := repo.UpdateSet(ctx, scopeId, h, item.GetVersion(), dbMask)
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(ctx, err, op, errors.WithMsg("unable to update host set"))
+	}
+	if rowsUpdated == 0 {
+		return nil, nil, nil, handlers.NotFoundErrorf("Host Set %q doesn't exist or incorrect version provided.", req.GetId())
+	}
+	var hl []host.Host
+	for _, h := range hosts {
+		hl = append(hl, h)
```

Comment:
You could just return `hosts` instead of `hl` here -- usually we do this because we're `toProto`-ing the values.

Alternately, replace with `hl := append([]host.Host, hosts...)` or so.

Reply:
hosts is of type *plugin.Host and we are needing to return a slice of host.Host.